subprojects/android-test/test-report/src/main/kotlin/com/avito/android/test/report/ReportImplementation.kt

```diff
@@ -113,39 +113,45 @@ class ReportImplementation(
         screenshot: FutureValue<RemoteStorage.Result>?,
     ) = methodExecutionTracing("registerIncident") {
         val currentState = getCastedState<ReportState.Initialized>()
-        val type = exception.determineIncidentType()
+
         if (currentState.incident == null) {
-            val incidentChain = IncidentChain.Impl(
-                customViewPresenters = setOf(
-                    TestCaseIncidentPresenter(),
-                    RequestIncidentPresenter(),
-                    ResourceIncidentPresenter(),
-                    ResourceManagerIncidentPresenter(),
-                    AppCrashIncidentPresenter()
-                ),
-                fallbackPresenter = FallbackIncidentPresenter()
-            ).toChain(exception)
+            if (screenshot != null) {
+                incidentFutureUploads.add(screenshot)
+            }
 
+            val type = exception.determineIncidentType()
+            val chainFactory = createIncidentChainFactory()
             val incidentToAdd = Incident(
                 type = type,
-                chain = incidentChain,
+                chain = chainFactory.toChain(exception),
                 timestamp = timeProvider.nowInSeconds(),
                 entryList = emptyList(),
                 trace = exception.formatStackTrace()
             )
-
-            if (screenshot != null) {
-                incidentFutureUploads.add(screenshot)
-            }
-            addText("Threads dump", ThreadDumper.getThreadDump())
-
-            onIncident.invoke(exception)
-
+            addTroubleshootingEntries()
             currentState.incident = incidentToAdd
             afterIncident(incidentToAdd)
+        } else {
+            logger.warn("Fail to register incident. Incident already exist", exception)
         }
     }
 
+    private fun createIncidentChainFactory() = IncidentChainFactory.Impl(
+        customViewPresenters = setOf(
+            TestCaseIncidentPresenter(),
+            RequestIncidentPresenter(),
+            ResourceIncidentPresenter(),
+            ResourceManagerIncidentPresenter(),
+            AppCrashIncidentPresenter()
+        ),
+        fallbackPresenter = FallbackIncidentPresenter()
+    )
+
+    private fun addTroubleshootingEntries() {
+        addText("Threads dump", ThreadDumper.getThreadDump())
+        addText("Main looper dump", MainLooperDumper.getDump())
```

Comment:
Do you have an example of how it looks?

Reply:
Now only, I can't test it fast now. I look sources and decide that is enough

Reply:
Ok, I'm just curious.